symfony/plugins/orangehrmCorePlugin/Service/CsvDataImportService.php

```diff
@@ -22,24 +22,42 @@
 
 use OrangeHRM\Core\Exception\DaoException;
 use OrangeHRM\Core\Import\CsvDataImportFactory;
+use OrangeHRM\Core\Traits\Service\TextHelperTrait;
 
 class CsvDataImportService
 {
+    use TextHelperTrait;
+
     /**
      * @param string $fileContent
      * @param string $importType
+     * @param array $headerValues
      * @return int
      * @throws DaoException
      */
     public function import(string $fileContent, string $importType, array $headerValues): int
     {
         $factory = new CsvDataImportFactory();
         $instance = $factory->getImportClassInstance($importType);
+
+        $stream = fopen('php://memory', 'r+');
```

Comment:
@samanthajayasinghe please check this when you have time

Reply:
This is the previous code.
```
$lines = explode("\n", $fileContent);
$employeesDataArray = array_map('str_getcsv', $lines);
```
This code caused an issue when line breaks were present in cells of the CSV. But if we are using `str_getcsv` it is necessary to explode by `\n` since that's how the next row can be identified. This is why we switched to `fgetcsv` as it can distinguish between a new row and preserves line breaks in the cells as well. 

`fgetcsv` does not accept string but needs a resource. So the solution we opted for was to open a stream in php memory and create a temporary file for `fgetcsv` to use